lambdas/es/indexer/document_queue.py

```diff
@@ -193,10 +211,59 @@ def send_all(self):
             verify_certs=True,
             connection_class=RequestsHttpConnection
         )
+
+    def _filter_and_delete_packages(self, elastic):
+        """handle package hard delete"""
+        logger_ = get_quilt_logger()
+        true_docs = []
+        for doc in self.queue:
+            pointer_file = doc.get("pointer_file")
+            # handle hard package delete outside of the bulk operation
+            # TODO: what does this do for deleting delete markers?
+            if doc["_op_type"] == "delete" and pointer_file and not doc.get("delete_marker"):
+                index = doc.get("_index")
+                assert index.endswith(PACKAGE_INDEX_SUFFIX), f"Refuse to delete non-package: {doc}"
+                handle = doc.get("handle")
+                assert handle, "Cannot delete package without handle"
+                # no try/except because failure to delete, or trying to delete things
+                # that aren't in ES, does not throw
+                deletes = elastic.delete_by_query(
+                    index=index,
+                    body={
+                        "query": {
+                            "bool": {
+                                "must": [
+                                    # use match (not term) because some of these fields are analyzed
+                                    {"match": {"handle": handle}},
+                                    {"match": {"pointer_file": pointer_file}},
+                                    {"match": {"delete_marker": False}},
+                                ]
+                            }
+                        }
+                    },
+                    # we delete synchronously, so don't let it linger too long
+                    timeout='20s'
```

Comment:
What happens if this times out?

Reply:
We don't do the delete. I could do this async.